Increased TSM-buffer size to 16 MiB

lowBudgetHypothermia opened the pull request:
The TSM-buffer size was increased to 16 MiB. As a consquence of this the TSM buffer has to be allocated on the heap for ltsmc.c and common.c to prevent segmentation faults.

ChristianTackeGSI commented on src/lib/common.c:
```diff
@@ -152,7 +152,7 @@ int crc32file(const char *filename, uint32_t *crc32result)
+	unsigned char* buf = malloc(TSM_BUF_LENGTH); // reserve memory on heap because stack too small
 
 	file = fopen(filename, "r");
 	if (file == NULL) {
```
Remember to put a 
```c
free(buf);
```
before the
```c
return rc;
```
or the
```c
return rc_minor;
```

ChristianTackeGSI commented on src/lib/common.c:
```diff
@@ -180,12 +181,15 @@ int crc32file(const char *filename, uint32_t *crc32result)
 	if (rc_minor) {
 		rc_minor = -errno;
 		CT_ERROR(rc_minor, "fclose failed on '%s'", filename);
+		free(buf);
 
 		return rc_minor;
 	}
 
 	*crc32result = crc32sum;
 
+	free(buf);
+
 	return rc;
 }
```
What about "falling through" in the last case? Might make things nicer?

and instead of inventing a new `rc_minor`, maybe just re-use `rc`.

```cpp
    rc = fclose(file);
	if (rc) {
		rc = -errno;
		CT_ERROR(rc, "fclose failed on '%s'", filename);
	}

	*crc32result = crc32sum;

	free(buf);

	return rc;
}
```

ChristianTackeGSI reviewed (commented):
Looks good!

ChristianTackeGSI reviewed (approved):
Looks good for now. :-)

I assume this was tested somewhat?

ChristianTackeGSI commented on src/lib/common.c:
```diff
@@ -180,11 +187,16 @@ int crc32file(const char *filename, uint32_t *crc32result)
 	if (rc_minor) {
 		rc_minor = -errno;
 		CT_ERROR(rc_minor, "fclose failed on '%s'", filename);
+		if (buf) free(buf);
```
Maybe move the `if (buf) free(buf);` above the `int rc_minor;`?

Something like this?

```c
if (buf) free(buf);
buf = NULL;

int rc_minor = fclose(file);
```

ChristianTackeGSI replied:
Concerns were raised in private that this could clobber some return values / `errno`. I appreciate the thoughtfulness about errno preservation in general.

Well.

1. > The free() function returns no value, and preserves errno.

   (from "man 3 free" on my Linux System)

   If anyone has a reference to an implementation or documentation indicating that `free()` can modify errno, please share it.

2. The `free()` is just below the while loop.

   - What errno should it clobber there? By `feof()`? Well according to my manpages, it does not set errno either.
   - The `fclose()` (where we want to preserve `errno`!) is just below the `free()`.

ChristianTackeGSI commented on src/lib/common.c:
```diff
@@ -174,8 +181,9 @@ int crc32file(const char *filename, uint32_t *crc32result)
 	} while (!feof(file));
 
-	int rc_minor;
+	if (buf) free(buf);
```
Maybe… just in case:

```suggestion
	if (buf) free(buf);
	buf = NULL;
```

lowBudgetHypothermia replied:
done

ChristianTackeGSI commented on src/lib/common.c:
```diff
@@ -174,8 +181,9 @@ int crc32file(const char *filename, uint32_t *crc32result)
+	if (buf) free(buf);
 
+	int rc_minor;
 	rc_minor = fclose(file);
```
Maybe?

```suggestion
	int rc_minor = fclose(file);
```

lowBudgetHypothermia replied:
done

ChristianTackeGSI commented on src/ltsmc.c:
```diff
@@ -493,7 +493,7 @@ int main(int argc, char *argv[])
 		}
 
-		char buf[TSM_BUF_LENGTH] = {0};
+		char* buf = malloc(TSM_BUF_LENGTH);
```
Please also check the return value here.

It might be NULL, right?

lowBudgetHypothermia replied:
done

ChristianTackeGSI commented on src/ltsmc.c:
```diff
@@ -515,9 +515,12 @@ int main(int argc, char *argv[])
 			CT_ERROR(errno, "tsm_fclose failed");
 
+		free(buf);
+
```
Maybe move this directly under the end of the while loop?

lowBudgetHypothermia replied:
done

ChristianTackeGSI commented on src/ltsmc.c:
```diff
@@ -515,9 +515,12 @@ int main(int argc, char *argv[])
 		goto cleanup_tsm;
 	}
 
+
```
Why adding an empty line here?

lowBudgetHypothermia replied:
Was introduced by accident. Should be removed in newest commit for `src/ltsmc.c`

ChristianTackeGSI reviewed (commented):
This looks good now!

We're at 8 commits. Please consider a rebase / squash / etc. and combine things into fewer commits?

TBH I don't know, what would be a good granularity, because after all its still only increasing and switching to malloc. All the rest is closely related cleanup / error checking, etc.

Commit:
Increased TSM-buffer size to 16 MiB

The TSM-buffer size was increased to 16 MiB. As a consquence of
this the TSM buffer has to be allocated on the heap for ltsmc.c
and common.c to prevent segmentation faults. A condition was
added in function crc32file() for setting crc32result.
